nifi-nar-bundles/nifi-cybersecurity-bundle/nifi-cybersecurity-processors/src/main/java/org/apache/nifi/processors/cybersecurity/CompareFuzzyHash.java

```diff
@@ -0,0 +1,358 @@
+                "by the <Hash Attribute Name> property. Note that: 'XXX' gets replaced with the <Hash Attribute Name>"),
+        @WritesAttribute(attribute = "XXXX.N.similarity", description = "The similarity score between this flowfile" +
+                "and the its match of the same number N. Note that: 'XXX' gets replaced with the <Hash Attribute Name>")})
+
+public class CompareFuzzyHash extends AbstractFuzzyHashProcessor {
+    public static final AllowableValue singleMatch = new AllowableValue(
+            "single",
+            "single",
+            "Send FlowFile to matched after the first match above threshold");
+    public static final AllowableValue multiMatch = new AllowableValue(
+            "multi-match",
+            "multi-match",
+            "Iterate full list of hashes before deciding to send FlowFile to matched or unmatched");
+
+    public static final PropertyDescriptor HASH_LIST_FILE = new PropertyDescriptor.Builder()
+            .name("HASH_LIST_FILE")
+            .displayName("Hash List source file")
+            .description("Path to the file containing hashes to be validated against")
+            .required(true)
+            .addValidator(StandardValidators.FILE_EXISTS_VALIDATOR)
+            .build();
+
+    // Note we add a PropertyDescriptor HASH_ALGORITHM and ATTRIBUTE_NAME from parent class
+
+
+
+    public static final PropertyDescriptor MATCH_THRESHOLD = new PropertyDescriptor.Builder()
+            // Note that while both TLSH and SSDeep seems to return int, we treat them as double in code.
+            // The rationale behind being the expectation that other algorithms thatmay return double values
+            // may be added to the processor later on.
+            .name("MATCH_THRESHOLD")
+            .displayName("Match threshold")
+            .description("The similarity score must exceed or be equal to in order for" +
+                    "match to be considered true. Refer to Additional Information for differences between TLSH " +
+                    "and SSDEEP scores and how they relate to this property.")
+            .required(true)
+            .addValidator(StandardValidators.NUMBER_VALIDATOR)
+            .build();
+
+    public static final PropertyDescriptor MATCHING_MODE = new PropertyDescriptor.Builder()
+            .name("MATCHING_MODE")
+            .displayName("Matching mode")
+            .description("The ")
+            .required(true)
+            .allowableValues(singleMatch,multiMatch)
+            .defaultValue(singleMatch.getValue())
+            .build();
+
+    public static final Relationship REL_MATCH = new Relationship.Builder()
+            .name("Matched")
+            .description("Any FlowFile that is successfully matched to an existing hash will be sent to this Relationship.")
+            .build();
+
+    public static final Relationship REL_NON_MATCH = new Relationship.Builder()
+            .name("non-match")
```

Comment:
Good call. renamed to "found", "not found" as per GeoEnrichIP and also added "failure"